simple_salesforce/api.py

```diff
@@ -442,11 +443,15 @@ def __init__(
         * sf_instance -- the domain of the instance of Salesforce to use
         * sf_version -- the version of the Salesforce API to use
         * proxies -- the optional map of scheme to proxy server
+        * session -- Custom requests session, created in calling code. This
+                     enables the use of requests Session features not otherwise
+                     exposed by simple_salesforce.
         """
         self.session_id = session_id
         self.name = object_name
-        self.request = requests.Session()
-        self.request.proxies = proxies
+        self.request = session or requests.Session()
```

Comment:
I see the inconsistent naming between this and the `Salesforce` object (`request` here but `session` in `Salesforce`). I would much rather see `session` used consistently between the two as it's what it's actually a reference to an instance of. Also, parameters and attributes will also be consistent.